core/src/main/java/com/graphhopper/routing/lm/LandmarkStorage.java

```diff
@@ -549,7 +549,10 @@ boolean initActiveLandmarks(int fromNode, int toNode, int[] activeLandmarkIndice
         if (subnetworkFrom <= UNCLEAR_SUBNETWORK || subnetworkTo <= UNCLEAR_SUBNETWORK)
             return false;
         if (subnetworkFrom != subnetworkTo) {
-            throw new ConnectionNotFoundException("Connection between locations not found. Different subnetworks " + subnetworkFrom + " vs. " + subnetworkTo, new HashMap<String, Object>());
+            Map<String, Object> details = new HashMap<String, Object>();
+            details.put("subnetwork-from", subnetworkFrom);
+            details.put("subnetwork-to", subnetworkTo);
```

Comment:
For json we should always use underscore

Reply:
Why would we need the subnetwork IDs parsable - what would be the use case?

Reply:
> Why would we need the subnetwork IDs parsable - what would be the use case?

The idea was that it would be possible to differentiate between the "different subnetwork" exception and the general "connection not found" exception. Both have different reasons and I think it is important for the client to be able to understand the difference and tell the user about it. But you are right, I think it might actually be better to have a different Exception class here. I know, we are adding a massive number of classes here, but I think there is a huge benefit for the client here, making it easier for users to help themselves, by receiving a useful, translated error message. Will add a new class here :).

Reply:
> I think it is important for the client to be able to understand the difference and tell the user about it

Ok, maybe the reason property is a good compromise if users want more details and we do not need to introduce new exception classes.


> In this case, the different subnetwork warning is created during our import

Where would we throw this in the import?

Reply:
> Where would we throw this in the import?

Sorry I think I described this a bit weird :). What I meant is that we create the separation during the import, in contrast to blocked areas.